packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts

```diff
@@ -791,4 +795,44 @@ export class QueryBuilderProjectionState {
     }
     return undefined;
   }
+
+  get usedPropertyNodeIds(): string[] {
```

Comment:
This makes sense now. However, im worried about the fact that we maintain in parallel 2 pieces of logic to compute node Id. Is there a way we share the utility between this and explorer state so if one changes, they would not go out of sync?

Reply:
To clarify further, what I meant is more like we come up with a good way to ensure the way we generate node ID here is the same way we generate the node ID when we do `new QueryBuilderExplorerTreeSubTypeNodeData(...)`. So for these, let's create some shared utility methods in `ExplorerState` to generate part of the ID.

Also, let's create a test for this, so we know these 2, `get usedPropertyNodeIds()` and `QueryBuilderExplorerState` always generate IDs the same way.